spec/datadog/tracing/sampling/span/matcher_spec.rb

```diff
@@ -0,0 +1,122 @@
+        { pattern: '?web', input: 'web1', expected: false },
+        { pattern: 'web?one', input: 'web-one', expected: true },
+        { pattern: 'web?one', input: 'webone', expected: false },
+        { pattern: 'web?one', input: 'web-site-one', expected: false },
+        { pattern: 'web?site?one', input: 'web-site-one', expected: true },
+        { pattern: 'web?site?one', input: 'web-nice-one', expected: false },
+      ],
+      'mixed * and ? pattern' => [
+        { pattern: '*web?', input: 'pre-web1', expected: true },
+        { pattern: '*web?', input: 'web', expected: false },
+        { pattern: '?web*', input: '1web-post', expected: true },
+        { pattern: '?web*', input: 'web', expected: false },
+        { pattern: 'web?one*', input: 'web-one-post', expected: true },
+        { pattern: 'web?one*', input: 'webone', expected: false },
+        { pattern: 'web*one?', input: 'web-one1', expected: true },
+        { pattern: 'web*one?', input: 'webne1', expected: false },
+        { pattern: 'web*site?one', input: 'web--site one', expected: true },
+        { pattern: 'web*site?one', input: 'web--siteone', expected: false },
+        { pattern: 'web?site*one', input: 'web-site  one', expected: true },
+        { pattern: 'web?site*one', input: 'web-sitene', expected: false },
+      ]
+    }.each do |scenario, fixtures|
+      context "for '#{scenario}'" do
+        fixtures.each do |pattern:, input:, expected:|
+          context "with pattern '#{pattern}' and input '#{input}'" do
+            context 'matching on span name' do
+              let(:matcher) { described_class.new(name_pattern: pattern) }
+              let(:span_name) { input }
+
+              it "does #{'not ' unless expected}match" do
+                is_expected.to eq(expected)
+              end
+            end
+
+            context 'matching on span service' do
+              let(:matcher) { described_class.new(service_pattern: pattern) }
+              let(:span_service) { input }
+
+              it "does #{'not ' unless expected}match" do
+                is_expected.to eq(expected)
+              end
+            end
+
+            context 'matching on span name and service' do
+              context 'with the same matching scenario for both fields' do
+                let(:matcher) { described_class.new(name_pattern: pattern, service_pattern: pattern) }
+                let(:span_name) { input }
+                let(:span_service) { input }
+
+                it "does #{'not ' unless expected}match" do
+                  is_expected.to eq(expected)
+                end
+              end
+            end
+          end
+        end
+      end
+    end
+
+    context 'matching on span name and service' do
+      let(:matcher) { described_class.new(name_pattern: name_pattern, service_pattern: service_pattern) }
+      let(:span_name) { input }
+      let(:span_service) { input }
+      let(:name_pattern) { input }
+      let(:service_pattern) { input }
```

Comment:
This is never used, and can be removed (it actually doesn't work, if I'm looking at it right)

Reply:
Left over from copy/paste. Luckily (un-luckily?) all these `let`s are overridden in each test case.
Removed these unused `let`s.